libs/vscode-ui/components/src/lib/select/select.component.spec.ts

```diff
@@ -1,21 +1,54 @@
+import { Component, ViewChild } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
 import { OptionType } from '@angular/cli/models/interface';
 import { Option, OptionComponent } from '@nx-console/schema';
 import { SelectComponent } from './select.component';
 
+const initialValue = 'test';
+const mockOption: Option = {
+  name: 'style',
+  description: 'The file extension to be used for style files.',
+  type: OptionType.String,
+  component: OptionComponent.Select,
+  aliases: [],
+  itemTooltips: {
+    test: 'testLabel'
+  },
+  items: [initialValue, 'other', 'values']
+};
+
+@Component({
+  template: `
+    <form [formGroup]="formGroup">
+      <nx-console-select [field]="field"></nx-console-select>
+    </form>
+  `
+})
+class ParentFormComponent {
+  field = mockOption;
+  formGroup = this.fb.group({[this.field.name]: initialValue});
+  @ViewChild(SelectComponent, {static: true}) selectComponent: SelectComponent;
+
+  constructor(private fb: FormBuilder) {}
+}
 describe('SelectComponent', () => {
+  let fixture: ComponentFixture<ParentFormComponent>;
+  let parent: ParentFormComponent;
   let component: SelectComponent;
-  const mockOption: Option = {
-    name: 'style',
-    description: 'The file extension to be used for style files.',
-    type: OptionType.String,
-    component: OptionComponent.Select,
-    aliases: [],
-    itemTooltips: {
-      test: 'testLabel'
-    }
-  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ParentFormComponent],
+      imports: [ReactiveFormsModule]
+    }).compileComponents();
+  }));
+
   beforeEach(() => {
-    component = new SelectComponent();
```

Comment:
I have never understood why I sometimes see Angular component specs without TestBed. Totally fine I guess when it's just a presentational component with no dependencies.

Reply:
This is unrelated to the review I am just curious about your opinion on this. In my view,  across Angular projects there are a lot of component tests that gain no benefit from TestBed. TestBed also adds code to your test making it longer in terms of lines to read and time taken to run.. If you have a component spec file that isn't interacting with the template I prefer to use no TestBed.  When you do want to test the template, I have found storybook/cypress is a better way to do it than using TestBed. 

This is the related ngconf video on this topic: https://www.youtube.com/watch?v=7JucMlrs3dQ

Reply:
I can see your point. These tests wouldn't have benefitted from TestBed as they were. Storybook/Cypress really can be more beneficial for testing the template. 

...I broke the existing tests by adding the ControlContainer dependency and wanted to add a test target to get these running with the rest of them. Ran into some blockers though.